### PR TITLE
Upgrade internal ingress API calls to v1;

### PR DESCRIPTION
**Pull request by ycliuhw**
*Upgrade internal ingress API calls to v1;*

Driveby fix: 
- Upgrade internal CRD and webhooks API calls to v1;

## Checklist

 - [ ] ~Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR~
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed~
 - [x] Comments answer the question of why design decisions were made

## QA steps

```console
$ juju deploy cs:~juju/mariadb-k8s-3
$ juju deploy cs:~juju/mediawiki-k8s-4 --config juju-external-hostname=172.16.104.2.xip.io
$ juju relate mediawiki-k8s:db mariadb-k8s:server
$ juju expose mediawiki-k8s
$ mkubectl get ing -nt1 mediawiki-k8s -o yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/rewrite-target: ""
    ingress.kubernetes.io/ssl-passthrough: "false"
    ingress.kubernetes.io/ssl-redirect: "false"
    kubernetes.io/ingress.allow-http: "false"
  creationTimestamp: "2021-01-28T04:03:06Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: juju
    app.kubernetes.io/name: mediawiki-k8s
    juju-controller-uuid: 9b96268b-2f5f-4a0b-8b5b-64f381846e15
    juju-model-uuid: 0520eed6-ba7d-4646-8d45-e780c4682a01
  name: mediawiki-k8s
  namespace: t1
  resourceVersion: "2180057"
  selfLink: /apis/networking.k8s.io/v1/namespaces/t1/ingresses/mediawiki-k8s
  uid: 9b021114-7656-47d8-a6ba-198e6a66402b
spec:
  ingressClassName: public
  rules:
  - host: 172.16.104.2.xip.io
    http:
      paths:
      - backend:
          service:
            name: mediawiki-k8s
            port:
              number: 80
        path: /
        pathType: ImplementationSpecific
status:
  loadBalancer:
    ingress:
    - ip: 127.0.0.1

$ juju remove-application mariadb-k8s -m k1:t1 --destroy-storage --force
$ juju remove-application mediawiki-k8s -m k1:t1 --destroy-storage --force
$ mkubectl get ing -nt1
No resources found in t1 namespace.
```

## Documentation changes

No

## Bug reference

No

### CODE REVIEW

**Comment by ycliuhw** [comment]
[upgrade ingress to v1 for `juju expose` and other internal API calls](https://trello.com/c/UGq3BWx6/1356-upgrade-ingress-to-v1-for-juju-expose-and-other-internal-api-calls)

File: caas/kubernetes/provider/admissionregistration.go [hunk 1]
```diff
@@ -135,74 +134,43 @@ func (k *kubernetesClient) EnsureMutatingWebhookConfiguration(cfg *admissionregi
 // nil.
 func (k *kubernetesClient) ensureMutatingWebhookConfigurationV1beta1(cfg *admissionregistrationv1beta1.MutatingWebhookConfiguration) (func(), error) {
 	cleanUp := func() {}
-	out, err := k.createMutatingWebhookConfiguration(cfg)
+	api := k.client().AdmissionregistrationV1beta1().MutatingWebhookConfigurations()
```
Comment (wallyworld) [inline caas/kubernetes/provider/admissionregistration.go:137]: This should just be v1 right?

**Review by wallyworld — APPROVED** [review]
This works with exiting deployments that were created using the v1beta apis?

File: caas/kubernetes/provider/admissionregistration.go (hunk 1 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/admissionregistration.go:137]: `ensureMutatingWebhookConfigurationV1beta1` is for handling `v1beta1` webhook spec provided in charm's podspec

**Comment by ycliuhw** [comment]
> This works with exiting deployments that were created using the v1beta apis?

Yes, we only changed the `list` and `delete` API calls to v1 which works for both versions.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**